src/rfq/protocols/hashflow/state.rs

```diff
@@ -45,7 +45,12 @@ impl ProtocolSim for HashflowState {
     }
 
     fn spot_price(&self, _base: &Token, _quote: &Token) -> Result<f64, SimulationError> {
-        todo!()
+        // Hashflow's levels are sorted by price, so the first level represents the best price.
+        self.levels
```

Comment:
also please check if the tokens match what we expected! Remember that Hashflow is _special_ and supports swaps only in one direction